tests/Unit/Http/Controllers/TestControllerTest.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+namespace Tests\Unit\Http\Controllers;
+
+use Tests\TestCase;
+use App\Http\Controllers\TestController;
+use App\Repositories\Test\TestRepositoryInterface;
+use Illuminate\Database\Eloquent\Collection;
+use Mockery as M;
+use Illuminate\Http\Request;
+use Illuminate\Http\RedirectResponse;
+use App\Models\Test;
+
+
+class TestControllerTest extends TestCase
+{
+    protected $testMock;
+    protected $testController;
+
+    public function setUp(): void
+    {
+        $this->testMock = M::mock(TestRepositoryInterface::class);
+        $this->testController = new TestController($this->testMock);
+        parent::setUp();
+    }
+
+    public function tearDown(): void
+    {
+        unset($this->testController);
+        M::close();
+        parent::tearDown();
+    }
+
+    public function testIndexFunction()
+    {
+        $this->testMock
+            ->shouldReceive('getAllTests')
+            ->once()
+            ->andReturn([]);
+        $result= $this->testController->index();
+        $data = $result->getData();
+        $this->assertEquals('history', $result->getName());
+        $this->assertArrayHasKey('tests', $data);
+    }
+
+    public function testCreateFunction()
+    {
+        $this->testMock
+            ->shouldReceive('getAllTypes')
+            ->once()
+            ->andReturn([]);
+        $this->testMock
+            ->shouldReceive('getAllDates')
+            ->once()
+            ->andReturn(new Collection());
+        $result = $this->testController->create();
+        $data = $result->getData();
+        $this->assertEquals('tests', $result->getName());
+        $this->assertArrayHasKey('types', $data);
+        $this->assertArrayHasKey('dates', $data);
+    }
+
+    public function testStoreFunctionWithEnglishLanguage()
```

Comment:
giải thích function này

Reply:
$this->assertArrayHasKey('dates', $data): kiểm tra xem array $data có key nào là dates ko ạ
testStoreFunctionWithEnglishLanguage() : test function store với case là ngôn ngữ english ạ